.github/workflows/commit.yml

```diff
@@ -58,7 +58,8 @@ jobs:
   test:
     name: Test
     needs: [generate-matrixes]
-    runs-on: ubuntu-20.04
+    if: github.event_name == 'pull_request'
```

Comment:
We only run the testing step on PRs because we re-implement the build/test in the `distribute` step below, as well as the sbom/vuln scan creation.

The SBOM/Vuln scans from the PR will be uploaded to the PR, but the SBOM/Vuln scans from the `distribute` step will be uploaded to the release. In theory they are the same or very similar, but there may be important differences so we maintain that by recreating them during `distribute`